### PR TITLE
Provide API to overload AppThemeProvider

### PR DESCRIPTION
**Pull request by minkimcello**
## Motivation

To address #7103 - This PR is to introduce a way for users to be able to specify their own `ThemeProvider` for the default Backstage frontend app.

## Approach

Added optional `ThemeProvider` as a property of `AppComponent`. By default `createApp()` will use Backstage's [`AppThemeProvider`](https://github.com/thefrontside/backstage/blob/mk/theme-provider/packages/core-app-api/src/app/AppThemeProvider.tsx#L71-L97) if one isn't provided.

```js
import { createApp } from '@backstage/core-app-api';
import { MyProvider } from './provider';

const MyApp = createApp({
  components: {
    ThemeProvider: MyProvider
    ...
  }
});
```

## Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/wild-eyes-roll.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/core-app-api': minor
```
Comment (Rugvip) [inline .changeset/wild-eyes-roll.md:2]: Should be a `patch` since this is not a breaking change and we're on 0.x

**Review by Rugvip — COMMENTED** [review]
Thank you, looks good! 👍 

Just a minor changeset nit

File: .changeset/wild-eyes-roll.md (hunk 1 above)
Comment (minkimcello) [reply .changeset/wild-eyes-roll.md]: 👍 changed to `patch`

**Review by Rugvip — APPROVED** [review]
:shipit:

**Comment by minkimcello** [comment]
@Rugvip I followed the instructions output in the terminal (from `yarn build:api-reports:only --ci --docs`) about copying/pasting the api reports from the cache to local files. Was it okay for me to do that?

**Comment by Rugvip** [comment]
@minkimcello yep they're supposed to be committed, although `yarn build:api-reports` is enough too 😁

**Review by freben — APPROVED** [review]

**Pull request merged**